src/hit/api/evaluator.cpp

```diff
@@ -85,11 +85,12 @@ namespace hit {
             LOG_AND_THROW_STREAM("Inputs to add must have the same scale: " << log2(ct1.scale()) << " bits != "
                                                                             << log2(ct2.scale()) << " bits");
         }
-        if (ct1.he_level() != ct2.he_level()) {
+        if (ct1.bootstrapped() == ct2.bootstrapped() && ct1.he_level() != ct2.he_level()) {
             LOG_AND_THROW_STREAM("Inputs to add must be at the same level: " << ct1.he_level()
                                                                              << " != " << ct2.he_level());
         }
         add_inplace_internal(ct1, ct2);
+        ct1.bootstrapped_ |= ct2.bootstrapped();
```

Comment:
Why boostrapped state can be `or |`?

Reply:
It's a boolean, and the idea is that binary operators can operate on one input that has already been bootstrapped and one input that has not yet been bootstrapped. So we have to define `bootstrapped` for the result. I've defined it so that if at least one of the inputs has `bootstrapped == true`, then the output also has `bootstrapped == true`.